doc/Internals.md

```diff
@@ -460,5 +460,7 @@ So if you need to trigger events in theia:
  - Register events with the emitter.event function
  - Fire events with emitter.fire(event)
 
+# Preferences
 
+You can check out [this page](Preferences.md) to learn more about how preferences work within Theia.
 
```

Comment:
How about a `doc/internals` directory, with one file per subject?

Also, do we want to make a distinction between "developer documentation", things that somebody writing plugins for Theia should know, and "internals", things that Theia developers need to know but somebody only writing plugins doesn't necessarily need to know?